src/status_im/contexts/wallet/wallet_connect/transactions.cljs

```diff
@@ -15,40 +29,116 @@
 
 (defn- format-tx-hex-values
   "Due to how status-go expects hex values, we should remove the extra 0s in transaction hex values e.g. 0x0f -> 0xf"
-  [tx]
+  [tx f]
   (let [tx-keys [:gasLimit :gas :gasPrice :nonce :value :maxFeePerGas :maxPriorityFeePerGas]]
     (reduce (fn [acc tx-key]
               (if (and (contains? tx tx-key)
                        (not (nil? (get tx tx-key))))
-                (update acc tx-key strip-hex-prefix)
+                (update acc tx-key f)
                 acc))
             tx
             tx-keys)))
 
+;; QUESTION: should we remove the nonce from the original tx so it's always set by status-go, so that
+;; we maintain the order of tx?
+;;
+;; Metamask seems to do so:
+;; https://docs.metamask.io/wallet/how-to/send-transactions#nonce
```

Comment:
Does the backend maintain a nonce ? Or expects the UI to send it. If the backend maintains it, we should rely on the backend so we don't run into a broken nonce state.

Reply:
if not provided in the tx, the backend will compute the next nonce, see [here](https://github.com/status-im/status-go/blob/898dedd59450499d733e5da4a6cf512379268a2a/transactions/transactor.go#L100). Are there cases where we would have to respect the nonce coming from the dapp? If not, will remove it and always compute it. 

cc: @stefandunca

Reply:
I think the backend should compute it always. We should completely reject nonces received from the outside world. A motivated enough actor can misuse it.